frame/nfts/src/lib.rs

```diff
@@ -939,21 +965,27 @@ pub mod pallet {
 				ensure!(permitted, Error::<T, I>::NoPermission);
 			}
 
-			details.approved = Some(delegate);
+			let now = frame_system::Pallet::<T>::block_number();
+			let deadline = maybe_deadline.map(|d| d.saturating_add(now));
+
+			details
+				.approvals
+				.try_insert(delegate.clone(), deadline.clone())
```

Comment:
```suggestion
				.try_insert(delegate.clone(), deadline)
```
to make clippy happy :)